dev/com.ibm.ws.concurrent.mp_fat_jakarta/test-applications/MPContextProp1_3_EE9_App/src/concurrent/mp/fat/v13/ee9/web/MPContextProp1_3_EE9_TestServlet.java

```diff
@@ -163,10 +163,19 @@ public void testFailureStackShowsCallerOfGetNow() throws Exception {
             fail("Got result " + result + " for a failed future");
         } catch (CancellationException x) {
             boolean foundInStack = false;
-            for (StackTraceElement line : x.getStackTrace())
+            for (StackTraceElement line : x.getStackTrace()) {
                 foundInStack |= "testFailureStackShowsCallerOfGetNow".equals(line.getMethodName());
-            if (!foundInStack || x.getCause() != CANCELLATION_X_CAUSED_BY_ARRAY_X.getCause())
+            }
+            // In Java 23, the pre-set failedFuture CancellationException gets wrapped in another CancellationException,
+            // so check if there is another cause and it is equal to the one we originally set as the failedFuture cause.
+            if (!foundInStack ||
+                !(x instanceof CancellationException) ||
+                (x.getCause() != CANCELLATION_X_CAUSED_BY_ARRAY_X.getCause() &&
+                 (x.getCause().getCause() == null ||
+                  !(x.getCause() instanceof CancellationException) ||
+                  x.getCause().getCause() != CANCELLATION_X_CAUSED_BY_ARRAY_X.getCause()))) {
```

Comment:
The cause should be compared with CancellationException to allow for Java 23, not the top level exception which is already compared via the `catch`.  This might work:

```suggestion
            if (!foundInStack ||
                x.getCause() instanceof CancellationException
                    ? x.getCause().getCause() != CANCELLATION_X_CAUSED_BY_ARRAY_X.getCause() // Java 23+
                    : x.getCause() != CANCELLATION_X_CAUSED_BY_ARRAY_X.getCause()) { // Java 22 or earlier
```